File: RotateArray/RotateArray.java
class RotateArray {
    public void rotate(int[] nums, int k) {
        int[] copy = new int[nums.length];
        System.arraycopy(nums, 0, copy, 0, nums.length);
        for (int i = 0; i < nums.length; i++) {
            nums[(i + k % nums.length) % nums.length] = copy[i];
        }
    }
}

File: RotateArray/RotateArray.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        # nums = nums[:-k % len(nums)]
        sec = nums[:-k % len(nums)]
        print(sec)
        del nums[:-k % len(nums)]
        nums.extend(sec)
